xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiCacheStore.java

```diff
@@ -318,14 +329,15 @@ public void onEvent(Event event, Object source, Object data)
     public void invalidate(XWikiDocument document)
     {
         String key = document.getKey();
+        this.cacheLoader.invalidate(key, k -> {
+            if (getCache() != null) {
```

Comment:
Would be simpler to add this null check in `#invalidateCache` and call that one here too no?

Reply:
I kept the difference of this one having `null` checks vs. the other not having `null` checks, but I don't know the background why this difference exists. We can also introduce `null` checks everywhere.

Reply:
> Would be simpler to add this null check in `#invalidateCache` and call that one here too no?

I did exactly that now and also removed the special case for delete as storing `false` in the exists cache feels not really thread-safe.